bin/dicomwebscp.js

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env node
+const StaticWado = require('../src')
+const Stats = require('../src/stats');
+
+const dcmjsDimse = require('dcmjs-dimse');
+const { Server, Scp } = dcmjsDimse;
+const { CEchoResponse, CFindResponse, CStoreResponse } = dcmjsDimse.responses;
+const {
+  Status,
+  PresentationContextResult,
+  RejectResult,
+  RejectSource,
+  RejectReason,
+  TransferSyntax,
+  SopClass,
+  StorageClass,
+} = dcmjsDimse.constants;
+
+const port = 11112;
+
+const AcceptedTransferSyntax = {
+  '1.2.840.10008.1.2': 'ImplicitVRLittleEndian',
+  '1.2.840.10008.1.2.1': 'ExplicitVRLittleEndian',
+  '1.2.840.10008.1.2.5': 'RleLossless',
+  '1.2.840.10008.1.2.4.50': 'JpegBaseline',
+  '1.2.840.10008.1.2.4.70': 'JpegLossless',
+  '1.2.840.10008.1.2.4.80': 'JpegLsLossless',
+  '1.2.840.10008.1.2.4.100': 'mpeg',
+  '1.2.840.10008.1.2.4.101': 'mpeg',
+  '1.2.840.10008.1.2.4.102': 'h264',
+  '1.2.840.10008.1.2.4.103': 'h264',
+};
+
+const defaults = {
+  isStudyData: true,
+  isGroup: true,
+};
+let i=0;
+
+class DcmjsDimseScp extends Scp {
+  constructor(socket, opts) {
+    super(socket, opts);
+    this.association = undefined;
+    console.log('New importer', ++i);
+    this.importer = new StaticWado(defaults);
+    this.importer.i = i;
+  }
+
+  // Handle incoming association requests
+  associationRequested(association) {
+    console.log("Association requested", association);
+    this.association = association;
+    // // Evaluate calling/called AET and reject association, if needed
+    // if (this.association.getCallingAeTitle() !== 'SCU') {
+    //   this.sendAssociationReject(
+    //     RejectResult.Permanent,
+    //     RejectSource.ServiceUser,
+    //     RejectReason.CallingAeNotRecognized
+    //   );
+    //   return;
+    // }
+
+    // Optionally set the preferred max PDU length
+    this.association.setMaxPduLength(65536);
+
+    const contexts = association.getPresentationContexts();
+    contexts.forEach((c) => {
+      const context = association.getPresentationContext(c.id);
+      if (
+        context.getAbstractSyntaxUid() === SopClass.Verification ||
+        Object.values(StorageClass).includes(context.getAbstractSyntaxUid())
+      ) {
+        const transferSyntaxes = context.getTransferSyntaxUids();
+        transferSyntaxes.forEach((transferSyntax) => {
+          if (AcceptedTransferSyntax[transferSyntax]) {
```

Comment:
What about not accepted transferSyntax? Maybe we should log something...

Reply:
No, logging something would leave a gazillion logs.  This is the wrong way to do this - the right way is to look to see if the most desired transfer syntax is available, and then proceed down the list, then if nothing is available, THEN log a message.  Going to tweak this to work properly

Reply:
Fixed this by choosing the desired tsuids in order, and then writing a message only on failure.